contracts/ContinuousTokenFactory.sol

```diff
@@ -2,9 +2,193 @@
 
 pragma solidity >=0.6.0 <0.7.0;
 
+import "@openzeppelin/contracts/math/SafeMath.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
+import "./ERC20Managed.sol";
+import "./Curve.sol";
 
 contract ContinuousTokenFactory {
+  using SafeMath for uint256;
 
+  ERC20 public reserveToken;
+
+  struct ContinuousToken {
+    address token;
+    address curve;
+  }
+
+  mapping(bytes32 => ContinuousToken) public continuousTokens;
+
+  constructor(address _reserveToken) public {
+    require(
+      _reserveToken != address(0),
+      "ContinuousTokenFactory: no reserve token ERC20 address"
+    );
+    reserveToken = ERC20(_reserveToken);
+  }
+
+  event NewContinuousToken(
+    bytes32 id,
+    address token,
+    string name,
+    address curve
+  );
+
+  function newContinuousToken(
+    bytes32 id, 
+    string memory name, 
+    string memory symbol,
+    address curve
+  ) public returns (bool) {
+    require(!exists(id), "ContinuousTokenFactory: invalid id");
+    // require(Curve(curve).price(0, 1), "ContinuousTokenFactory: invalid curve");
```

Comment:
Long term we can make our interfaces support ERC-165 https://eips.ethereum.org/EIPS/eip-165